### PR TITLE
py-numpy: conflict with gcc11 and switch master to main

### PR DESCRIPTION
**Pull request by manuelakuhn**
This fixes the syntax of the conflict between numpy 1.21.0 and gcc11 so that the clingo concretizer recognizes it (see below).
In addition the upstream master branch was renamed to main.

Concretization before:
```
$ spack spec py-numpy
Input spec
--------------------------------
py-numpy

Concretized
--------------------------------
py-numpy@1.21.0%gcc@11.1.0+blas+lapack patches=873745d7b547857fcfec9cae90b09c133b42a4f0c23b6c2d84cf37e2dd816604 arch=linux-fedora32-haswell
    ^openblas@0.3.15%gcc@11.1.0~bignuma~consistent_fpcsr~ilp64+locking+pic+shared threads=none arch=linux-fedora32-haswell
        ^perl@5.34.0%gcc@11.1.0+cpanm+shared+threads arch=linux-fedora32-haswell
            ^berkeley-db@18.1.40%gcc@11.1.0+cxx~docs+stl patches=b231fcc4d5cff05e5c3a4814f6a5af0e9a966428dc2176540d2c05aff41de522 arch=linux-fedora32-haswell
            ^gdbm@1.19%gcc@11.1.0 arch=linux-fedora32-haswell
                ^readline@8.1%gcc@11.1.0 arch=linux-fedora32-haswell
                    ^ncurses@6.2%gcc@11.1.0~symlinks+termlib abi=none arch=linux-fedora32-haswell
                        ^pkgconf@1.7.4%gcc@11.1.0 arch=linux-fedora32-haswell
    ^py-cython@0.29.22%gcc@11.1.0 arch=linux-fedora32-haswell
        ^py-setuptools@50.3.2%gcc@11.1.0 arch=linux-fedora32-haswell
            ^python@3.8.10%gcc@11.1.0+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93189bc278fbc37a50ed7f183bd8aaf249a8e1670a465f0db6bb4f8cf87 arch=linux-fedora32-haswell
                ^bzip2@1.0.8%gcc@11.1.0~debug~pic+shared arch=linux-fedora32-haswell
                    ^diffutils@3.7%gcc@11.1.0 arch=linux-fedora32-haswell
                        ^libiconv@1.16%gcc@11.1.0 arch=linux-fedora32-haswell
                ^expat@2.3.0%gcc@11.1.0+libbsd arch=linux-fedora32-haswell
                    ^libbsd@0.11.3%gcc@11.1.0 arch=linux-fedora32-haswell
                        ^libmd@1.0.3%gcc@11.1.0 arch=linux-fedora32-haswell
                ^gettext@0.21%gcc@11.1.0+bzip2+curses+git~libunistring+libxml2+tar+xz arch=linux-fedora32-haswell
                    ^libxml2@2.9.10%gcc@11.1.0~python arch=linux-fedora32-haswell
                        ^xz@5.2.5%gcc@11.1.0~pic libs=shared,static arch=linux-fedora32-haswell
                        ^zlib@1.2.11%gcc@11.1.0+optimize+pic+shared arch=linux-fedora32-haswell
                    ^tar@1.34%gcc@11.1.0 arch=linux-fedora32-haswell
                ^libffi@3.3%gcc@11.1.0 patches=26f26c6f29a7ce9bf370ad3ab2610f99365b4bdd7b82e7c31df41a3370d685c0 arch=linux-fedora32-haswell
                ^openssl@1.1.1k%gcc@11.1.0~docs+systemcerts arch=linux-fedora32-haswell
                ^sqlite@3.35.5%gcc@11.1.0+column_metadata+fts~functions~rtree arch=linux-fedora32-haswell
                ^util-linux-uuid@2.36.2%gcc@11.1.0 arch=linux-fedora32-haswell
```
-> still uses gcc together with numpy 1.21.0

Concretization after:
```
$ spack spec py-numpy
Input spec
--------------------------------
py-numpy

Concretized
--------------------------------
py-numpy@1.21.0%gcc@10.2.1+blas+lapack patches=873745d7b547857fcfec9cae90b09c133b42a4f0c23b6c2d84cf37e2dd816604 arch=linux-fedora32-haswell
    ^openblas@0.3.15%gcc@10.2.1~bignuma~consistent_fpcsr~ilp64+locking+pic+shared threads=none arch=linux-fedora32-haswell
        ^perl@5.34.0%gcc@10.2.1+cpanm+shared+threads arch=linux-fedora32-haswell
            ^berkeley-db@18.1.40%gcc@10.2.1+cxx~docs+stl patches=b231fcc4d5cff05e5c3a4814f6a5af0e9a966428dc2176540d2c05aff41de522 arch=linux-fedora32-haswell
            ^gdbm@1.19%gcc@10.2.1 arch=linux-fedora32-haswell
                ^readline@8.1%gcc@10.2.1 arch=linux-fedora32-haswell
                    ^ncurses@6.2%gcc@10.2.1~symlinks+termlib abi=none arch=linux-fedora32-haswell
                        ^pkgconf@1.7.4%gcc@10.2.1 arch=linux-fedora32-haswell
    ^py-cython@0.29.22%gcc@10.2.1 arch=linux-fedora32-haswell
        ^py-setuptools@50.3.2%gcc@10.2.1 arch=linux-fedora32-haswell
            ^python@3.8.10%gcc@10.2.1+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93189bc278fbc37a50ed7f183bd8aaf249a8e1670a465f0db6bb4f8cf87 arch=linux-fedora32-haswell
                ^bzip2@1.0.8%gcc@10.2.1~debug~pic+shared arch=linux-fedora32-haswell
                    ^diffutils@3.7%gcc@10.2.1 arch=linux-fedora32-haswell
                        ^libiconv@1.16%gcc@10.2.1 arch=linux-fedora32-haswell
                ^expat@2.3.0%gcc@10.2.1+libbsd arch=linux-fedora32-haswell
                    ^libbsd@0.11.3%gcc@10.2.1 arch=linux-fedora32-haswell
                        ^libmd@1.0.3%gcc@10.2.1 arch=linux-fedora32-haswell
                ^gettext@0.21%gcc@10.2.1+bzip2+curses+git~libunistring+libxml2+tar+xz arch=linux-fedora32-haswell
                    ^libxml2@2.9.10%gcc@10.2.1~python arch=linux-fedora32-haswell
                        ^xz@5.2.5%gcc@10.2.1~pic libs=shared,static arch=linux-fedora32-haswell
                        ^zlib@1.2.11%gcc@10.2.1+optimize+pic+shared arch=linux-fedora32-haswell
                    ^tar@1.34%gcc@10.2.1 arch=linux-fedora32-haswell
                ^libffi@3.3%gcc@10.2.1 patches=26f26c6f29a7ce9bf370ad3ab2610f99365b4bdd7b82e7c31df41a3370d685c0 arch=linux-fedora32-haswell
                ^openssl@1.1.1k%gcc@10.2.1~docs+systemcerts arch=linux-fedora32-haswell
                ^sqlite@3.35.5%gcc@10.2.1+column_metadata+fts~functions~rtree arch=linux-fedora32-haswell
                ^util-linux-uuid@2.36.2%gcc@10.2.1 arch=linux-fedora32-haswell
```
-> downgrades gcc
resp.:
```
$ spack spec py-numpy %gcc@11.1.0
Input spec
--------------------------------
py-numpy%gcc@11.1.0

Concretized
--------------------------------
py-numpy@1.20.3%gcc@11.1.0+blas+lapack patches=873745d7b547857fcfec9cae90b09c133b42a4f0c23b6c2d84cf37e2dd816604 arch=linux-fedora32-haswell
    ^openblas@0.3.15%gcc@11.1.0~bignuma~consistent_fpcsr~ilp64+locking+pic+shared threads=none arch=linux-fedora32-haswell
        ^perl@5.34.0%gcc@11.1.0+cpanm+shared+threads arch=linux-fedora32-haswell
            ^berkeley-db@18.1.40%gcc@11.1.0+cxx~docs+stl patches=b231fcc4d5cff05e5c3a4814f6a5af0e9a966428dc2176540d2c05aff41de522 arch=linux-fedora32-haswell
            ^gdbm@1.19%gcc@11.1.0 arch=linux-fedora32-haswell
                ^readline@8.1%gcc@11.1.0 arch=linux-fedora32-haswell
                    ^ncurses@6.2%gcc@11.1.0~symlinks+termlib abi=none arch=linux-fedora32-haswell
                        ^pkgconf@1.7.4%gcc@11.1.0 arch=linux-fedora32-haswell
    ^py-cython@0.29.22%gcc@11.1.0 arch=linux-fedora32-haswell
        ^py-setuptools@50.3.2%gcc@11.1.0 arch=linux-fedora32-haswell
            ^python@3.8.10%gcc@11.1.0+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93189bc278fbc37a50ed7f183bd8aaf249a8e1670a465f0db6bb4f8cf87 arch=linux-fedora32-haswell
                ^bzip2@1.0.8%gcc@11.1.0~debug~pic+shared arch=linux-fedora32-haswell
                    ^diffutils@3.7%gcc@11.1.0 arch=linux-fedora32-haswell
                        ^libiconv@1.16%gcc@11.1.0 arch=linux-fedora32-haswell
                ^expat@2.3.0%gcc@11.1.0+libbsd arch=linux-fedora32-haswell
                    ^libbsd@0.11.3%gcc@11.1.0 arch=linux-fedora32-haswell
                        ^libmd@1.0.3%gcc@11.1.0 arch=linux-fedora32-haswell
                ^gettext@0.21%gcc@11.1.0+bzip2+curses+git~libunistring+libxml2+tar+xz arch=linux-fedora32-haswell
                    ^libxml2@2.9.10%gcc@11.1.0~python arch=linux-fedora32-haswell
                        ^xz@5.2.5%gcc@11.1.0~pic libs=shared,static arch=linux-fedora32-haswell
                        ^zlib@1.2.11%gcc@11.1.0+optimize+pic+shared arch=linux-fedora32-haswell
                    ^tar@1.34%gcc@11.1.0 arch=linux-fedora32-haswell
                ^libffi@3.3%gcc@11.1.0 patches=26f26c6f29a7ce9bf370ad3ab2610f99365b4bdd7b82e7c31df41a3370d685c0 arch=linux-fedora32-haswell
                ^openssl@1.1.1k%gcc@11.1.0~docs+systemcerts arch=linux-fedora32-haswell
                ^sqlite@3.35.5%gcc@11.1.0+column_metadata+fts~functions~rtree arch=linux-fedora32-haswell
                ^util-linux-uuid@2.36.2%gcc@11.1.0 arch=linux-fedora32-haswell
```
-> downgrades numpy

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-numpy/package.py [hunk 1]
```diff
@@ -115,8 +116,9 @@ class PyNumpy(PythonPackage):
     patch('check_executables4.patch', when='@1.14.0:1.15.4')
     patch('check_executables5.patch', when='@:1.13.3')
 
-    # https://github.com/numpy/numpy/releases/tag/v1.21.0
-    conflicts('%gcc@11.1', when='@1.21.0')
+    # version 1.21.0 runs into an infinit loop during printing
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-numpy/package.py:120]:
```suggestion
    # version 1.21.0 runs into an infinite loop during printing
```

**Review by adamjstewart — DISMISSED** [review]

**Comment by manuelakuhn** [comment]
Any idea why the style check fails now? They also affect lines now I did not touch. When I do `spack style` all checks run through as clean.

**Comment by adamjstewart** [comment]
```
var/spack/repos/builtin/packages/py-numpy/package.py:7: [I100] Import statements are in the wrong order. 'import platform' should be before 'from spack import *' and in a different group.
var/spack/repos/builtin/packages/py-numpy/package.py:7: [I201] Missing newline between import groups. 'import platform' is identified as Stdlib and 'from spack import *' is identified as Application.
```
You just need to re-order the imports. I recently added `flake8-import-order` to our `spack style` tests. You'll need to install `flake8-import-order` to reproduce the error messages locally. You can also use a tool like `isort` to automatically reorder your imports.

**Comment by manuelakuhn** [comment]
Unfortunately the local style check still did not fail even with `flake8-import-order` installed.
I have changed the import order nonetheless.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**